packages/premint-sdk/src/premint-api.ts

```diff
@@ -0,0 +1,684 @@
+        premintConfig,
+        chainId: this.chain.id,
+      }),
+    });
+
+    if (checkSignature) {
+      const [isValidSignature] = await publicClient.readContract({
+        abi: zoraCreator1155PremintExecutorImplABI,
+        address: this.getExecutorAddress(),
+        functionName: "isValidSignature",
+        args: [collection, premintConfig, signature],
+      });
+      if (!isValidSignature) {
+        throw new Error("Invalid signature");
+      }
+    }
+
+    const apiData = {
+      collection,
+      premint: encodePremintForAPI(premintConfig),
+      chain_name: this.network.zoraBackendChainName,
+      signature: signature,
+    };
+
+    let premint = await this.retries(() =>
+      this.post(`${ZORA_API_BASE}signature`, apiData),
+    );
+
+    return {
+      zoraUrl: `https://${
+        this.network.isTestnet ? "testnet." : ""
+      }zora.co/collect/${
+        this.network.zoraPathChainName
+      }:${verifyingContract}/premint-${uid}`,
+      uid,
+      verifyingContract,
+      premint,
+    };
+  }
+
+  private async retries<T>(
+    tryFn: () => T,
+    maxTries: number = 3,
+    atTry: number = 1,
+  ): Promise<T> {
+    try {
+      return await tryFn();
+    } catch (err: any) {
+      if (err instanceof BadResponse) {
+        if (atTry >= maxTries) {
+          await wait(500);
+          return await this.retries(tryFn, maxTries, atTry++);
+        }
+      }
+      throw err;
+    }
+  }
+
+  /**
+   * Create premint
+   *
+   * @param settings Settings for the new premint
+   * @param settings.account Account to sign the premint with. Taken from walletClient if none passed in.
+   * @param settings.collection Collection information for the mint
+   * @param settings.token Mint argument settings, optional settings are overridden with sensible defaults.
+   * @param settings.publicClient Public client (optional) – instantiated if not passed in with defaults.
+   * @param settings.walletClient Required wallet client for signing the premint message.
+   * @param settings.executionSettings Execution settings for premint options
+   * @param settings.executionSettings.deleted If this UID should be deleted. If omitted, set to false.
+   * @param settings.executionSettings.uid the UID to use – optional and retrieved as a fresh UID from ZORA by default.
+   * @param settings.checkSignature if the signature should have a pre-flight check. Not required but helpful for debugging.
+   * @returns premint url, uid, newContractAddress, and premint object
+   */
+  async createPremint({
+    account,
+    collection,
+    token,
+    publicClient,
+    walletClient,
+    executionSettings,
+    checkSignature = false,
+  }: {
+    account: Address;
+    checkSignature?: boolean;
+    walletClient: WalletClient;
+    collection: PremintResponse["collection"];
+    token: MintArgumentsSettings;
+    publicClient?: PublicClient;
+    executionSettings?: {
+      deleted?: boolean;
+      uid?: number;
+    };
+  }) {
```

Comment:
could you add the explicit return type here? it's not clear from reading the code what it is (you have to look at the implementation of `signAndSubmitPremint`